### PR TITLE
Forbid transfer to themselves for Withdrawal nft

### PR DESCRIPTION
**Pull request by folkyatina**

### CODE REVIEW

File: contracts/0.8.9/WithdrawalQueueERC721.sol [hunk 1]
```diff
@@ -191,8 +193,8 @@ contract WithdrawalQueueERC721 is IERC721Metadata, WithdrawalQueue {
         delete _getTokenApprovals()[_requestId];
         request.owner = payable(_to);
 
-        _getRequestsByOwner()[_to].add(_requestId);
         _getRequestsByOwner()[_from].remove(_requestId);
+        _getRequestsByOwner()[_to].add(_requestId);
```
Comment (TheDZhon) [inline contracts/0.8.9/WithdrawalQueueERC721.sol:197]: as a kind of safety measure, I would suggest checking the returns of `add` and `remove` methods of the OZ `EnumerableSet` here and there, and maybe assert on them

**Review by TheDZhon — APPROVED** [review]
The fix is essential :heart: 
Suggested a bit more hardening.

File: contracts/0.8.9/WithdrawalQueueERC721.sol (hunk 1 above)
Comment (folkyatina) [reply contracts/0.8.9/WithdrawalQueueERC721.sol:197]: I think we can. It will be the third ring of defence, but relatively cheap one.

**Review by TheDZhon — APPROVED** [review]
:rocket:

**Pull request merged**